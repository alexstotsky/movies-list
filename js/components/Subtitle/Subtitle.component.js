import React from 'react'
import { View, Text } from 'react-native'
import PropTypes from 'prop-types'

import { styleSheet } from '../../utils/commonPropTypes'

class SubTitle extends React.PureComponent {
  static propTypes = {
    subtitle: PropTypes.string.isRequired,
    containerStyle: styleSheet,
    subtitleStyle: styleSheet,
  }

  static defaultProps = {
    subtitle: '',
  }

  render() {
    const { subtitle, containerStyle, subtitleStyle } = this.props
    return (
      <View style={containerStyle}>
        <Text style={subtitleStyle}>
          {subtitle}
        </Text>
      </View>
    )
  }
}

export default SubTitle
