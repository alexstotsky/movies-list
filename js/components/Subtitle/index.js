import Subtitle from './Subtitle.component'

export default Subtitle
