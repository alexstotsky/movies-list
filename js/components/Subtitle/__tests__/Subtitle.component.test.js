import React from 'react'
import { shallow } from 'enzyme'

import Subtitle from '../Subtitle.component'

describe('Testing Subtitle component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Subtitle subtitle='' />,
    );
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ subtitle: undefined })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ subtitle: 'asd' })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ containerStyle: null })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ containerStyle: 55 })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ containerStyle: {} })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ subtitleStyle: null })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ subtitleStyle: 55 })
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ subtitleStyle: {} })
    expect(wrapper).toMatchSnapshot()
  });
});
