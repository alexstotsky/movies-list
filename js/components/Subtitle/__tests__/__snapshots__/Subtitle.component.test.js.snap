// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Subtitle component renders correctly 1`] = `
<View>
  <Text />
</View>
`;

exports[`Testing Subtitle component renders correctly 2`] = `
<View>
  <Text />
</View>
`;

exports[`Testing Subtitle component renders correctly 3`] = `
<View>
  <Text>
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 4`] = `
<View
  style={null}
>
  <Text>
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 5`] = `
<View
  style={55}
>
  <Text>
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 6`] = `
<View
  style={Object {}}
>
  <Text>
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 7`] = `
<View
  style={Object {}}
>
  <Text
    style={null}
  >
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 8`] = `
<View
  style={Object {}}
>
  <Text
    style={55}
  >
    asd
  </Text>
</View>
`;

exports[`Testing Subtitle component renders correctly 9`] = `
<View
  style={Object {}}
>
  <Text
    style={Object {}}
  >
    asd
  </Text>
</View>
`;
